### PR TITLE
Don't cascade maximized windows

### LINKED ISSUE #10335: When the current window is maximized, opening a new window should open the new one maximized
From **@50Wliu** on https://github.com/atom/atom/pull/10236#issuecomment-169855151

> After testing this out on Windows it looks like there needs to be some extra Windows-specific code: opening a new window that spawns from a maximized one should stay maximized and not cascade.

### PR DESCRIPTION
**Pull request by winstliu**
Fixes #10335

### CODE REVIEW

File: src/browser/atom-application.coffee [hunk 1]
```diff
@@ -369,6 +369,7 @@ class AtomApplication
   # Get the dimensions for opening a new window by cascading as appropriate to
   # the platform.
   getDimensionsForNewWindow: ->
+    return if (@focusedWindow() ? @lastFocusedWindow)?.isMaximized() and process.platform isnt 'darwin'
```
Comment (joshaber) [inline src/browser/atom-application.coffee]: I don't think we need to bother conditionalizing this since that's the right behavior for OS X, too.

**Comment by joshaber** [comment]
Awesome, thank you! Just one note.

**Comment by 50Wliu** [comment]
@joshaber Changed!

**Comment by joshaber** [comment]
Awesome, thanks again!

**Comment by joshaber** [comment]
What do you think about cherry picking this onto beta so that this is fixed when the cascading windows ships?

**Comment by 50Wliu** [comment]
> What do you think about cherry picking this onto beta so that this is fixed when the cascading windows ships?

Sure thing!

**Pull request merged**